src/components/CommentsForm.jsx

```diff
@@ -0,0 +1,100 @@
+import {  useState } from "react";
+import { useParams } from "react-router-dom";
+import { postComments } from "../utils/api";
+
+
+
+const CommentsForm =({commentsList, setCommentsList,})=>{
+const params =useParams();
+
+const articleId = params.article_id;
+
+const[userName, setUserName]=useState('tickle122')
+
+const[body, setBody]=useState("")
+const [err, setErr] =useState(null);
+
+const[message, setMessage]= useState("");
+
+
+
+const handleSubmit =(event)=>{
+    event.preventDefault();
+   
+    setUserName("tickle122");
+
+    const inputComment = {
+    
+    username: userName,
+         body: body,
+         votes:0,
+                    };
+                console.log(inputComment);
+                  setCommentsList((currentComments)=> {
+                    return ([inputComment, ...currentComments])})
+                    setMessage("We have updated your comment");
+                     setErr(null)
+ postComments(articleId, inputComment).catch((err) => {
+    if(body === ""){
+        setMessage("Please enter a comment")}
+        else{
+            setCommentsList((currentComments) => { 
+       (currentComments.shift()); })
```

Comment:
We should never mutate the state in these functions, you should make a copy of the currentComments and alter that instead 👍